src/public/Interfaces.ts

```diff
@@ -121,37 +120,33 @@ export interface IBindingDefinition {
      */
     type: string, 
     /**
-     * The direction of your binding ('in', 'out', or 'inout'), as defined in function.json.
+     * The direction of your binding, as defined in function.json.
      */
-    direction: string
+    direction: 'in' | 'out' | 'inout';
 }
 
-export interface ILog {
-    (...args: any[]): void;
-}
-  
 /**
  * Allows you to write streaming function logs. 
  */
-export interface ILogger extends ILog {
+export interface Logger {
+    /**
+     * Writes streaming function logs at the default trace level. 
+     */
+    (...args: any[]): void;
```

Comment:
Can this really take anything for the first argument? (like a structured log?) Or will the first argument be expected to be a string?

(Ditto for other methods)

Reply:
For the logs ones it's truly any (it looks like [util.format](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bb560329b92bbb6a2a085bb4244c937637595da2/types/node/v8/index.d.ts#L5667) used in WorkerChannel.ts takes types of "any").